project2/py/recipe.py

```diff
@@ -2,12 +2,19 @@
 
 
 class Recipe:
-    def __init__(self, ingredients, directions):
+    def __init__(self, ingredients, directions, methods, tools):
         self.ingredients = ingredients
         self.directions = directions
+        self.methods = methods
+        self.tools = tools
 
     def __str__(self):
         ings = utils.pmap(str, self.ingredients)
+        mthds = utils.pmap(str, self.methods)
```

Comment:
I believe these are strings already so is mapping to strings necessary? I had to map ingredients to str above since they were instances of `Ingredient`